### PR TITLE
Bugfix/rating updating

### LINKED ISSUE #74: Movie Page needs to show User Rating

### PR DESCRIPTION
**Pull request by Atos20**
#### What’s this PR do?  
 - it allows the user to see immediate changes made to movie ratings 
- it removes the error messages found when a given movie hasn't been rated by the user
- it displays the user's rating for the movie that the user has rated, but only when the user has logged in 
#### Where should the reviewer start?  
`APp.js` and `MovieCard.js`
#### How should this be manually tested?  
 run `npm start` then add and remove ratings and watch the changes happening in real-time
#### Any background context you want to provide?  
 We needed to display the changes made by the user in real-time, we figure out that we needed to make the proper network request to get the most updated information and the calling the `setState` method to re-render the necessary components.

#### What are the relevant tickets?  
 closes #74

### CODE REVIEW

**Comment by Atos20** [comment]
I resolved the conflicts, let me know if you have any questions.

File: src/MoviePage/MovieCard/MovieCard.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function MovieCard({movieDetails}) {
 
         <div className="box runtime-container">
           <h1 className="title rating-title">Rating</h1>
-          <p className="value average-rating">{movieDetails.average_rating}</p>
+          <p className="value average-rating">{movieDetails.average_rating.toFixed(1)}</p>
```
Comment (holladayian) [inline src/MoviePage/MovieCard/MovieCard.js:22]: What does `.toFixed(1)` do again?

File: src/App/App.js [hunk 2]
```diff
@@ -83,20 +82,16 @@ export class App extends Component{
...
-      // )
-      // .then(() => this.componentDidMount())
-      .then(ratedMovies => this.setState({ ratedMovies }))
-      )
+    } else if (!movieToRate) {
+      const newRating = {movie_id: this.state.movieID, rating: desiredRatingByUser.value}
+      const addNewRating = await fetcher.fetchCreateUserRating(this.state.userData.id, newRating)
+      const allUserRatings = await fetcher.fetchUserRatings(this.state.userData.id)
+      this.getMovieDetails(this.state.movieID)
+      this.setState({ratedMovies: allUserRatings})
```
Comment (holladayian) [inline src/App/App.js:94]: this.setState( {ratedMovies: allUserRatings} ) for readability! Also, we'll have to go back and lint the entire project. I know that is part of our grading rubric.

File: src/App/App.js [hunk 3]
```diff
@@ -83,12 +82,12 @@ export class App extends Component{
...
-   addRating = async (desiredRating) => {
-    const usersRating = this.state.ratedMovies.find(ratedMovie => ratedMovie.movie_id === this.state.movieID)
-    if (!desiredRating) {
+   addRating = async (desiredRatingByUser) => {
+    const movieToRate = this.state.ratedMovies.find(ratedMovie => ratedMovie.movie_id === this.state.movieID)
+    if (!desiredRatingByUser) {
       return alert('You have to select a rating to rate a movie!')
-    } else if (!usersRating) {
-      const newRating = {movie_id: this.state.movieID, rating: desiredRating.value}
+    } else if (!movieToRate) {
```
Comment (holladayian) [inline src/App/App.js]: Nice! I like these naming techniques a lot more.

**Comment by holladayian** [comment]
Looks good! Just a few clarifying questions. I'll merge.

File: src/MoviePage/MovieCard/MovieCard.js (hunk 1 above)
Comment (Atos20) [reply src/MoviePage/MovieCard/MovieCard.js:22]: it shortens or extends  a number to its first decimal.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed

File: src/App/App.js (hunk 2 above)
Comment (Atos20) [reply src/App/App.js:94]: oh shooot!!

**Pull request merged**